tests/bsim/bluetooth/host/l2cap/credits_seg_recv/src/main.c

```diff
@@ -0,0 +1,450 @@
+	int err;
+	struct bt_l2cap_le_chan *le_chan = &test_ctx.le_chan;
+
+	*le_chan = (struct bt_l2cap_le_chan){
+		.chan.ops = &ops,
+		.rx.mtu = L2CAP_MTU,
+		.rx.mps = BT_L2CAP_RX_MTU,
+	};
+
+	UNSET_FLAG(flag_l2cap_connected);
+
+	/* Credits can be given before requesting the connection and
+	 * after the 'connected' event. Credits given before connecting
+	 * are sent in the 'initial credits' field of the connection
+	 * request PDU.
+	 */
+	bt_l2cap_chan_give_credits(&le_chan->chan, 1);
+
+	err = bt_l2cap_chan_connect(conn, &le_chan->chan, 0x0080);
+	ASSERT(!err, "Error connecting l2cap channel (err %d)\n", err);
+
+	WAIT_FOR_FLAG_SET(flag_l2cap_connected);
+}
+
+static void connect_l2cap_ecred_channel(struct bt_conn *conn, void *data)
+{
+	int err;
+	struct bt_l2cap_le_chan *le_chan = &test_ctx.le_chan;
+	struct bt_l2cap_chan *chan_list[2] = {&le_chan->chan, 0};
+
+	*le_chan = (struct bt_l2cap_le_chan){
+		.chan.ops = &ops,
+		.rx.mtu = L2CAP_MTU,
+		.rx.mps = BT_L2CAP_RX_MTU,
+	};
+
+	UNSET_FLAG(flag_l2cap_connected);
+
+	/* Credits can be given before requesting the connection and
+	 * after the 'connected' event. Credits given before connecting
+	 * are sent in the 'initial credits' field of the connection
+	 * request PDU.
+	 */
+	bt_l2cap_chan_give_credits(&le_chan->chan, 1);
+
+	err = bt_l2cap_ecred_chan_connect(conn, chan_list, 0x0080);
+	ASSERT(!err, "Error connecting l2cap channel (err %d)\n", err);
+
+	WAIT_FOR_FLAG_SET(flag_l2cap_connected);
+}
+
+static void test_central_main(void)
+{
+	LOG_DBG("*L2CAP CREDITS Central started*");
+	int err;
+
+	/* Prepare tx_data */
+	for (size_t i = 0; i < sizeof(tx_data); i++) {
+		tx_data[i] = (uint8_t)i;
+	}
+
+	err = bt_enable(NULL);
+	ASSERT(err == 0, "Can't enable Bluetooth (err %d)\n", err);
+	LOG_DBG("Central Bluetooth initialized.");
+
+	connect_peripheral();
+
+	/* Connect L2CAP channels */
+	LOG_DBG("Connect L2CAP channels");
+	if (IS_ENABLED(CONFIG_BT_L2CAP_ECRED)) {
+		bt_conn_foreach(BT_CONN_TYPE_LE, connect_l2cap_ecred_channel, NULL);
+	} else {
+		bt_conn_foreach(BT_CONN_TYPE_LE, connect_l2cap_channel, NULL);
+	}
+
+	/* Send SDU_NUM SDUs to each peripheral */
+	test_ctx.tx_left = SDU_NUM;
+	l2cap_chan_send(&test_ctx.le_chan.chan, tx_data, sizeof(tx_data));
+
+	LOG_DBG("Wait until all transfers are completed.");
+	while (test_ctx.tx_left) {
+		k_msleep(100);
+	}
+
+	WAIT_FOR_FLAG_UNSET(is_connected);
+	LOG_DBG("Peripheral disconnected.");
+	PASS("L2CAP CREDITS Central passed\n");
+}
+
+static const struct bst_test_instance test_def[] = {{.test_id = "peripheral",
+						     .test_descr = "Peripheral L2CAP CREDITS",
+						     .test_post_init_f = test_init,
+						     .test_tick_f = test_tick,
+						     .test_main_f = test_peripheral_main},
+						    {.test_id = "central",
+						     .test_descr = "Central L2CAP CREDITS",
+						     .test_post_init_f = test_init,
+						     .test_tick_f = test_tick,
+						     .test_main_f = test_central_main},
+						    BSTEST_END_MARKER};
```

Comment:
This looks like the work of clang-format :D 

Suggest add `,` to the last value so that clang-format will format it in a more sane way.

Reply:
Thanks! :D
I too was a bit taken aback by what clang-format did.